include/oneapi/dpl/internal/binary_search_impl.h

```diff
@@ -87,9 +85,7 @@ upper_bound_impl(Policy&& policy, InputIterator1 start, InputIterator1 end, Inpu
                  InputIterator2 value_end, OutputIterator result, StrictWeakOrdering comp)
 {
     return oneapi::dpl::transform(policy, value_start, value_end, result,
-                                  [=](typename ::std::iterator_traits<InputIterator2>::reference val) {
-                                      return ::std::upper_bound(start, end, val, comp) - start;
-                                  });
+                                  [=](auto val) { return ::std::upper_bound(start, end, val, comp) - start; });
```

Comment:
Looks like it's not equivalent change to replace
`typename ::std::iterator_traits<InputIterator2>::reference val` to `auto`

Reply:
Actually, binary_search_impl.h was reverted.